pennylane/devices/qubit/measure.py

```diff
@@ -109,18 +109,24 @@ def get_measurement_function(
             if measurementprocess.obs.name == "SparseHamiltonian":
                 return csr_dot_products
 
-            if isinstance(measurementprocess.obs, Hamiltonian) or (
-                isinstance(measurementprocess.obs, Sum)
-                and measurementprocess.obs.has_overlapping_wires
-                and len(measurementprocess.obs.wires) > 7
-            ):
-                # Use tensor contraction for `Sum` expectation values with non-commuting summands
-                # and 8 or more wires as it's faster than using eigenvalues.
-
+            backprop_mode = math.get_interface(state) != "numpy"
```

Comment:
```suggestion
            backprop_mode = math.get_interface(state, *measurementprocess.obs.data) != "numpy"
```
Might also be something we need to consider. The state could still be numpy but the observable could have trainable parameters.

Reply:
I think it hasn't been a problem because we always [cast the state](https://github.com/PennyLaneAI/pennylane/blob/a60cf2074baae32d9b725c802886a16a0c7cdb62/pennylane/devices/default_qubit.py#L681) in the old DefaultQubit, but this will be good to have because we probably would have hit a bug in the future since we [aren't casting the state in new devices](https://github.com/PennyLaneAI/pennylane/blob/a60cf2074baae32d9b725c802886a16a0c7cdb62/pennylane/devices/qubit/simulate.py#L58) (yet, just need to pass the appropriate "like" value)

Reply:
i'll do this in a separate PR after to keep this one simple.